mzLib/Omics/IBioPolymer.cs

```diff
@@ -1,24 +1,25 @@
-﻿using Chemistry;
-using MassSpectrometry;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using Omics.Digestion;
+﻿using Omics.Digestion;
 using Omics.Modifications;
 
 namespace Omics
 {
     public interface IBioPolymer 
     {
         string Name { get; }
+        string FullName { get; }
         string BaseSequence { get; }
         int Length { get; }
         string DatabaseFilePath { get; }
         bool IsDecoy { get; }
         bool IsContaminant { get; }
         string Organism { get; }
         string Accession { get; }
+        /// <summary>
+        /// The list of gene names consists of tuples, where Item1 is the type of gene name, and Item2 is the nameof the specific molecule.
```

Comment:
https://www.uniprot.org/help/gene_name

Reply:
![image](https://github.com/smith-chem-wisc/mzLib/assets/16841846/e4c39753-eca0-47cd-b388-e7d2a1ffd784)

Reply:
when reading uniprot xml, the two fields that go in the tuple in this case are "primary" and "dpiA", which is the most common gene symbol. I believe that the other pairs of words are added as the tuple in the list. I'm not sure what is done with those values.

Reply:
Here is my attempt at a comment:
//This tuple originates from reading UniProt XML protein entries. It consists of two parts.
//The second part is the gene name, which is also often referred to as the gene symbol.
//The first part is a descriptor for the second part. For example, "primary" is a word used for the 
//first part to indicate that the gene symbol/name that follows is the name most widely accepted
//by the community. The word "synonym" in the first part would indicate that the gene name in the
//second part is an alternative. The word "ORF", which stands for open reading frame in the first part
//is also acceptable. Here, the second part will include a range, for example "AO055_05240"